osdep/macosx_touchbar.m

```diff
@@ -0,0 +1,202 @@
+        tbItem.slider.maxValue = 100.0f;
+        tbItem.target = self;
+        tbItem.action = @selector(seekbarChanged:);
+        tbItem.customizationLabel = @"Seek Bar";
+        self.touchbarSeekbar = tbItem.slider;
+        return tbItem;
+    } else if ([identifier isEqualToString:play]) {
+        NSCustomTouchBarItem *tbItem = [self tbButtonWithId:identifier
+                                                       name:@"Play Button"
+                                                      image:NSImageNameTouchBarPauseTemplate
+                                                     action:@selector(play:)];
+        self.touchbarPlay = (NSButton *)tbItem.view;
+        return tbItem;
+    } else if ([identifier isEqualToString:previousItem]) {
+        return [self tbButtonWithId:identifier
+                               name:@"Previous"
+                              image:NSImageNameTouchBarGoBackTemplate
+                             action:@selector(previousItem:)];
+    } else if ([identifier isEqualToString:nextItem]) {
+        return [self tbButtonWithId:identifier
+                               name:@"Next"
+                              image:NSImageNameTouchBarGoForwardTemplate
+                             action:@selector(nextItem:)];
+    } else if ([identifier isEqualToString:previousChapter]) {
+        return [self tbButtonWithId:identifier
+                               name:@"Previous Chapter"
+                              image:NSImageNameTouchBarSkipBackTemplate
+                             action:@selector(previousChapter:)];
+    } else if ([identifier isEqualToString:nextChapter]) {
+        return [self tbButtonWithId:identifier
+                               name:@"Next Chapter"
+                              image:NSImageNameTouchBarSkipAheadTemplate
+                             action:@selector(nextChapter:)];
+    } else if ([identifier isEqualToString:cycleAudio]) {
+        return [self tbButtonWithId:identifier
+                               name:@"Cycle Audio"
+                              image:NSImageNameTouchBarAudioInputTemplate
+                             action:@selector(cycleAudio:)];
+    } else if ([identifier isEqualToString:cycleSubtitle]) {
+        return [self tbButtonWithId:identifier
+                               name:@"Cycle Subtitle"
+                              image:NSImageNameTouchBarComposeTemplate
+                             action:@selector(cycleSubtitle:)];
+    } else if ([identifier isEqualToString:currentPosition]) {
+        NSCustomTouchBarItem *tbItem = [self tbTextWithId:identifier name:@"Current Position"];
+        self.touchbarCurrentPosition = (NSTextField *)tbItem.view;
+        return tbItem;
+    } else if ([identifier isEqualToString:timeLeft]) {
+        NSCustomTouchBarItem *tbItem = [self tbTextWithId:identifier name:@"Time Left"];
+        self.touchbarTimeLeft = (NSTextField *)tbItem.view;
+        return tbItem;
+    }
+    return nil;
+}
```

Comment:
It seems to be [Apple's preferred way](https://developer.apple.com/library/prerelease/content/samplecode/NSTouchBarCatalog/Listings/Objective_C_NSTouchBar_Catalog_TestViewControllers_PopoverViewController_m.html#//apple_ref/doc/uid/TP40017550-Objective_C_NSTouchBar_Catalog_TestViewControllers_PopoverViewController_m-DontLinkElementID_33) thought.
<img width="762" alt="screen shot 2017-03-02 at 13 32 28" src="https://cloud.githubusercontent.com/assets/1769603/23507422/4cea1f1a-ff4d-11e6-8b5c-5692f46010d2.png">